### PR TITLE
@tus/server: do not emit chunkFinished on error

### PR DESCRIPTION
**Pull request by fenos**
Related to https://github.com/tus/tus-node-server/issues/444 

If an error occurs during the write stream, we don't want to broadcast the chunkFinished event.
This should fix potential race conditions

### CODE REVIEW

File: packages/server/src/models/StreamSplitter.ts [hunk 1]
```diff
@@ -85,17 +85,20 @@ export class StreamSplitter extends stream.Writable {
...
       return
     }
 
     await this.fileHandle.close()
 
-    this.emit('chunkFinished', {
-      path: this.currentChunkPath,
-      size: this.currentChunkSize,
-    })
+    if (!err) {
```
Comment (Murderlon) [inline packages/server/src/models/StreamSplitter.ts:95]: Can we add an elaborate comment on why this is needed here?

**Review by Murderlon — COMMENTED** [review]
Thanks a lot for looking into this!

**Comment by fenos** [comment]
@Murderlon actually, I think a cleaner way to handle this is to have its own error handler function instead of mixing it into the `_finishChunk` function, for example:

```ts
  async _handleError() {
    if (this.fileHandle === null) {
      return
    }

    await this.fileHandle.close()
    this.currentChunkPath = null
    this.fileHandle = null
  }
```

It feels cleaner doing it this way, since we don't even need to advance the part to next chunk either.

**Review by Murderlon — APPROVED** [review]
Thank you!

**Comment by Murderlon** [comment]
> @Murderlon actually, I think a cleaner way to handle this is to have its own error handler function instead of mixing it into the `_finishChunk` function, for example:

Probably a bit cleaner yes. Let's add it.

**Comment by fenos** [comment]
@Murderlon Just sent the patch using its dedicated error handler function

**Comment by Murderlon** [comment]
Released https://github.com/tus/tus-node-server/releases/tag/%40tus%2Fserver%401.0.0-beta.6

**Pull request merged**